### ISSUE TITLE
Solver takes over 10 minutes, resulting in a failed Travis build

### ISSUE THREAD

**Issue by snoyberg** [haskell/cabal#4298]
You can see the Travis build here: https://travis-ci.org/yesodweb/yesod/jobs/199252979. Steps to reproduce:

1.   Clone the repo https://github.com/yesodweb/yesod at commit `9ee3c3707441963835be04d38b14831c6d9e4b29`.

2.   With GHC 8.0.1 and cabal-install 1.24.0.0, run:
    ```
    cabal install --dry-run --only-dependencies --enable-tests --enable-benchmarks --force-reinstalls --ghc-options=-O0 --reorder-goals --max-backjumps=-1 yesod-bin/ yesod-websockets/ yesod-test/ yesod-core/ yesod-sitemap/ yesod-newsfeed/ yesod-persistent/ yesod/ yesod-form/ yesod-eventsource/ yesod-auth-oauth/ yesod-static/ yesod-auth/
    ```
3.  I did this with an index last updated at 16:42 on 2017-02-07

Expected: the build plan can be calculated.

Actual: build plan took greater than 10 minutes, causing the Travis build to fail.

**Comment by phadej** [comment]
could you try `cabal-head`? it has much better solver.

**Comment by snoyberg** [comment]
Sure, I've just pushed a commit for that. Build should occur at:

https://travis-ci.org/yesodweb/yesod/builds/199293634

**Comment by 23Skidoo** [comment]
Seems to be fixed in `HEAD`.

**Comment by 23Skidoo** [comment]
@snoyberg Do you need a fix for this issue backported the 1.24 branch, or can you wait for the 2.0 release (which should be out at the same time as GHC 8.2)?

**Comment by snoyberg** [comment]
If it works with cabal-head, I can update the Travis job to just use that for 8.0.1, which should be fine overall. No need for this specific issue to cut a new release, but thank you. Worst case scenario, if HEAD turns out to be unstable, I'll just add this to an expected-failures case and leave the 8.0.1 testing to Stack.
